src/index.js

```diff
@@ -242,10 +395,10 @@ class ClientAuthOIDC {
    * clear any http-only session cookies.
    */
   logout () {
-    this.clearCurrentUser()
-
     let logoutEndpoint = this.providerEndSessionEndpoint()
 
+    this.clearCurrentCredentials()
```

Comment:
So, part of the reason I do it before, is that here `logout()` actually redirects the current page to the idp's `/logout` endpoint (so that the server can clear the browser's http-only cookie for that session).